### ISSUE TITLE
[TextInput] Programatically restrict TextInput values

### ISSUE THREAD

**Issue by alvaromb** [facebook/react-native#5128]
I want to programatically restrict ``TextInput`` values and make it support only numbers. I've tried to do this modifying the value inside the ``onChangeText`` event. The component state does get the changed value, but it is not reflected inside the ``TextInput`` component. I've created a RNPlay project to illustrate: https://rnplay.org/apps/v8Jyqw

In contrast, this is what the docs say about the ``value`` prop of the ``TextInput`` component:

> The value to show for the text input. TextInput is a controlled component, which means the native value will be forced to match this value prop if provided. For most uses this works great, but in some cases this may cause flickering - one common cause is preventing edits by keeping value the same. In addition to simply setting the same value, either set editable={false}, or set/update maxLength to prevent unwanted edits without flicker.

**Comment by feichngr** [comment]
Same for me.
I'm on 0.17 and TextInput on iOS still ignores the `value` prop after initial rendering (providing it a changed value from the parent component's state or forcing a rerender of the parent component with `forceUpdate` doesn't work).
I hoped this would be fixed with c8108b (which I think has landed in 0.17) but completely forgot to check.

**Comment by alvaromb** [comment]
Ok, let's see if @brentvatne can confirm that, @feichngr :)

**Comment by feichngr** [comment]
I just tried 0.18-rc and it works for me now.
I misread the git log, c8108bdbe13590660d7f575d0747e100ea7bb866 was not in 0.17, so that commit probably fixed it.

**Comment by hufeng** [comment]
@alvaromb @feichngr, react-native@0.15 I met the same problem. at last, I find this, for example
```javascript

import React, {TextInput} from 'react-native';
import TimerMixin from 'react-time-mixin';

const TextDemo = React.createClass({
  mixins: [TimerMixin],

 getInitialState() {
    return {value: ''};
 },

  render() {
    return (
       <TextInput 
            value={this.state.value} 
            onChangeText={(text) => {this.requestAnimationFrame(() => {this.setState({value: '/*what you want to set*/'})})}}/>
    );
  }
});

```

**Comment by brentvatne** [comment]
Yeah this should be fixed in c8108bdbe13590660d7f575d0747e100ea7bb866  :)
